core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts

```diff
@@ -1,26 +1,212 @@
 
-import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { HttpClientModule } from '@angular/common/http';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnInit,
+    Output,
+    ViewChild
+} from '@angular/core';
 
 import { ButtonModule } from 'primeng/button';
 import { DialogModule } from 'primeng/dialog';
 
+import { skip } from 'rxjs/operators';
+
 import { DotMessageService } from '@dotcms/data-access';
+import { DotCMSTempFile } from '@dotcms/dotcms-models';
+import {
+    DotDropZoneComponent,
+    DotMessagePipe,
+    DotSpinnerModule,
+    DropZoneFileEvent,
+    DropZoneFileValidity
+} from '@dotcms/ui';
+
+import { DotUiMessageComponent } from './components/dot-ui-message/dot-ui-message.component';
+import {
+    BINARY_FIELD_MODE,
+    BINARY_FIELD_STATUS,
+    DotBinaryFieldStore,
+    initialState
+} from './store/binary-field.store';
+
+import { UI_MESSAGE_KEYS, UiMessageI, getUiMessage } from '../../utils/binary-field-utils';
 
 @Component({
-    selector: 'dotcms-binary-field',
+    selector: 'dot-binary-field',
     standalone: true,
-    imports: [ButtonModule, DialogModule, MonacoEditorModule],
+    imports: [
+        CommonModule,
+        ButtonModule,
+        DialogModule,
+        DotDropZoneComponent,
+        MonacoEditorModule,
+        DotMessagePipe,
+        DotUiMessageComponent,
+        DotSpinnerModule,
+        HttpClientModule
+    ],
+    providers: [DotBinaryFieldStore],
     templateUrl: './binary-field.component.html',
     styleUrls: ['./binary-field.component.scss'],
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class BinaryFieldComponent {
-    private readonly dotMessageService = inject(DotMessageService);
-    labelCode = this.dotMessageService.get('contenttypes.content.edit.write.code');
-    visible = false;
+export class DotBinaryFieldComponent implements OnInit {
+    //Inputs
+    acceptedTypes: string[] = [];
+    @Input() set accept(accept: string) {
+        this.acceptedTypes = accept.split(',').map((type) => type.trim());
+    }
+
+    @Input() maxFileSize: number;
+    @Input() helperText: string;
+
+    @ViewChild('inputFile') inputFile: ElementRef;
+
+    @Output() tempFile = new EventEmitter<DotCMSTempFile>();
+
+    readonly dialogHeaderMap = {
+        [BINARY_FIELD_MODE.URL]: 'dot.binary.field.dialog.import.from.url.header',
+        [BINARY_FIELD_MODE.EDITOR]: 'dot.binary.field.dialog.create.new.file.header'
+    };
+    readonly BINARY_FIELD_STATUS = BINARY_FIELD_STATUS;
+    readonly BINARY_FIELD_MODE = BINARY_FIELD_MODE;
+    readonly vm$ = this.dotBinaryFieldStore.vm$;
+
+    constructor(
+        private readonly dotBinaryFieldStore: DotBinaryFieldStore,
+        private readonly dotMessageService: DotMessageService
+    ) {
+        this.dotBinaryFieldStore.setState(initialState);
```

Comment:
What but if we are doing this without required async data or anything needed from the component like an input this can be done in the store.